repokid/utils/dynamo_v2.py

```diff
@@ -0,0 +1,262 @@
+
+logger = logging.getLogger("repokid")
+
+
+def _empty_string_from_dynamo_replace(obj: T) -> T:
+    """
+    Traverse a potentially nested object and replace all Dynamo placeholders with actual empty strings
+
+    Args:
+        obj (object)
+
+    Returns:
+        object: Object with original empty strings
+    """
+    if isinstance(obj, dict):
+        return {k: _empty_string_from_dynamo_replace(v) for k, v in list(obj.items())}
+    elif isinstance(obj, list):
+        return [_empty_string_from_dynamo_replace(elem) for elem in obj]
+    else:
+        if isinstance(obj, str) and str == DYNAMO_EMPTY_STRING:
+            return ""
+        return obj
+
+
+def _empty_string_to_dynamo_replace(
+    obj: Union[Dict[str, Any], List[Any]]
+) -> Union[Dict[str, Any], List[Any]]:
+    """
+    Traverse a potentially nested object and replace all instances of an empty string with a placeholder
+
+    Args:
+        obj (object)
+
+    Returns:
+        object: Object with Dynamo friendly empty strings
+    """
+    if isinstance(obj, dict):
+        return {k: _empty_string_to_dynamo_replace(v) for k, v in list(obj.items())}
+    elif isinstance(obj, list):
+        return [_empty_string_to_dynamo_replace(elem) for elem in obj]
+    else:
+        try:
+            if str(obj) == "":
+                obj = DYNAMO_EMPTY_STRING
+        except UnicodeEncodeError:
+            obj = DYNAMO_EMPTY_STRING
+        return obj
+
+
+def _datetime_to_string_replace(
+    obj: Union[Dict[str, Any], List[Any]]
+) -> Union[Dict[str, Any], List[Any]]:
+    if isinstance(obj, dict):
+        return {k: _datetime_to_string_replace(v) for k, v in list(obj.items())}
+    elif isinstance(obj, list):
+        return [_datetime_to_string_replace(elem) for elem in obj]
+    else:
+        if isinstance(obj, datetime.datetime):
+            obj = obj.strftime("%Y-%m-%d %H:%M")
+        return obj
+
+
+def dynamo_get_or_create_table(
+    account_number: str,
+    session_name: str,
+    region: str,
+    endpoint: str,
+    assume_role: Optional[str] = "",
+) -> Table:
+    """
+    Create a new table or get a reference to an existing Dynamo table named 'repokid_roles' that will store data all
+    data for Repokid.  Return a table with a reference to the dynamo resource
+
+    Args:
+        account_number (string)
+        assume_role (string) optional
+        session_name (string)
+        region (string)
+        endpoint (string)
+
+    Returns:
+        dynamo_table object
+    """
+    if "localhost" in endpoint:
+        resource = boto3.resource(
+            "dynamodb", region_name="us-east-1", endpoint_url=endpoint
+        )
+    else:
+        resource = boto3_cached_conn(
+            "dynamodb",
+            service_type="resource",
+            account_number=account_number,
+            assume_role=assume_role or None,
+            session_name=session_name,
+            region=region,
+        )
+
+    for table in resource.tables.all():
```

Comment:
Oh this is cool, so you just create the table in code if it doesn't exist? Maybe we should do this in ConsoleMe.